hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcDuplexHandler.java

```diff
@@ -103,8 +103,8 @@ private void writeRequest(ChannelHandlerContext ctx, Call call, ChannelPromise p
         ctx.write(buf, withoutCellBlockPromise);
         ChannelPromise cellBlockPromise = ctx.newPromise();
         ctx.write(cellBlock, cellBlockPromise);
-        PromiseCombiner combiner = new PromiseCombiner();
-        combiner.addAll(withoutCellBlockPromise, cellBlockPromise);
+        PromiseCombiner combiner = new PromiseCombiner(ctx.executor());
+        combiner.addAll((ChannelFuture) withoutCellBlockPromise, cellBlockPromise);
```

Comment:
We need a cast is needed so we call a subclass method ?

Reply:
The method which takes Promises as parameters is deprecated.

https://github.com/netty/netty/blob/3e8e52725bbd9d999873880956bc8f241372daa8/common/src/main/java/io/netty/util/concurrent/PromiseCombiner.java#L123

So here we cast the first one to a future to let it call the method which takes Futures as parameters(Promise is a sub interface of Future).